smartapps/juano2310/jawbone-up-connect.src/jawbone-up-connect.groovy

```diff
@@ -349,67 +352,49 @@ def uninstalled() {
 }
 
 def pollChild(childDevice) {
-    def member = state.member
-    generatePollingEvents (member, childDevice)
+		def params = [
+      uri: 'https://jawbone.com',
+      path: '/nudge/api/users/@me/goals',
+      headers: ["Authorization": "Bearer ${state.JawboneAccessToken}" ]
+    ]
+    asynchttp_v1.get('responseGoals', params, childDevice)
+
+		def params = [
+			uri: 'https://jawbone.com',
+			path: '/nudge/api/users/@me/moves',
+			headers: ["Authorization": "Bearer ${state.JawboneAccessToken}" ]
+		]
+		asynchttp_v1.get('responseMoves', params, childDevice)
 }
 
-def generatePollingEvents (member, childDevice) {
-    // lets figure out if the member is currently "home" (At the place)
-    def urlgoals = "https://jawbone.com/nudge/api/users/@me/goals"
-    def urlmoves = "https://jawbone.com/nudge/api/users/@me/moves"
-    def urlsleeps = "https://jawbone.com/nudge/api/users/@me/sleeps"
-    def goals = null
-    def moves = null
- 	def sleeps = null
-    httpGet(uri: urlgoals, headers: ["Authorization": "Bearer ${state.JawboneAccessToken}" ]) {response ->
-        goals = response.data.data
-    }
-    httpGet(uri: urlmoves, headers: ["Authorization": "Bearer ${state.JawboneAccessToken}" ]) {response ->
-        moves = response.data.data.items[0]
-    }
-
-    try { // we are going to just ignore any errors
-        log.debug "Member = ${member.first}"
-        log.debug "Moves Goal = ${goals.move_steps} Steps"
-        log.debug "Moves = ${moves.details.steps} Steps"
+def generateInitialEvent (childDevice) {
+		childDevice?.generateSleepingEvent(false)
 
-        childDevice?.sendEvent(name:"steps", value: moves.details.steps)
-        childDevice?.sendEvent(name:"goal", value: goals.move_steps)
-    	//setColor(moves.details.steps,goals.move_steps,childDevice)
-    }
-    catch (e) {
-            // eat it
-    }
+		def params = [
+      uri: 'https://jawbone.com',
+      path: '/nudge/api/users/@me/goals',
+      headers: ["Authorization": "Bearer ${state.JawboneAccessToken}" ]
+    ]
+    asynchttp_v1.get('responseGoals', params, childDevice)
```

Comment:
before being async would a failure response cause the install to fail? And if it did, is that desired functionality? If not, or if we don't care, then I think this would be fine to make async.